static/skin/index.js

```diff
@@ -197,13 +197,63 @@
         }
     }
 
+    /* hack for library.kiwix.org magnet links (created by MirrorBrain)
+        See https://github.com/kiwix/container-images/issues/242 */
+    async function getFixedMirrorbrainMagnet(magnetLink) {
+        // parse as query parameters
+        const params = new URLSearchParams(
+            magnetLink.replaceAll('&amp;', '&').replace(/^magnet:/, ''));
+
+        const zimUrl = params.get('as'); // as= is fallback URL
+        // download metalink to build list of mirrored URLs
+        let mirrorUrls = [];
+        try {
+            const metalink = await fetch(`${zimUrl}.meta4`).then(response => {
+                return response.ok ? response.text() : '';
+            }).catch((_error) => '');
+            if (metalink) {
+                const parser = new DOMParser();
+                const doc = parser.parseFromString(metalink, "application/xml");
+                doc.querySelectorAll("url").forEach((node) => {
+                    if (node.hasAttribute("priority")) { // ensures its a mirror link
+                        mirrorUrls.push(node.innerHTML);
+                    }
+                });
+            }
+        } catch (err) {
+            // not a big deal, magnet will only contain primary URL
+            console.debug(`Failed to retrieve mirror links for ${zimUrl}`);
+        }
+
+        // set webseed (ws=) URL to primary download URL (redirects to mirror)
+        params.set('ws', zimUrl);
+        // if we got metalink mirror URLs, append them all
+        if (mirrorUrls) {
+            mirrorUrls.forEach((url) => {
+                params.append('ws', url);
+            });
+        }
+
+        params.set('xs', `${zimUrl}.torrent`);  // adding xs= to point to torrent URL
+
+        magnetLink = 'magnet:?';
+        // we don't want params.ToString()'s urlencoding
+        for (const [key, value] of params.entries()) {
+             magnetLink += `&${key}=${value}`;
+        }
```

Comment:
1. This produces a link of the form `magnet:?&...` - note the ampersand symbol immediately following the question mark symbol. Is that OK?

2. In our unit tests we have a ZIM file with an ampersand symbol in it, which suggests that such ZIM files occur in the wild. Won't URLs to such files pose a problem in the absence of URL-encoding?

Reply:
> This produces a link of the form magnet:?&... - note the ampersand symbol immediately following the question mark symbol. Is that OK?

Yes, it is. We could remove the first `&` but that's extra processing for something that's already a valid URI

> In our unit tests we have a ZIM file with an ampersand symbol in it, which suggests that such ZIM files occur in the wild. Won't URLs to such files pose a problem in the absence of URL-encoding?

Yes, actually the spec says it should be urlencoded. I had it previously and removed it when I encountered an issue (prob. unrelated). Let me fix that back

Reply:
Although spec is not clear about URLencoding some of the params, I chose to URL encode all now and tested that al of them are properly decoded in my test clients.

Please note that we dont publish any ZIM file with url-encodable characters ATM.